libbeat/docs/newbeat.asciidoc

```diff
@@ -146,203 +214,222 @@ NOTE: To be consistent with other Beats, you should append `beat` to your Beat n
-file. Here is an example configuration section for Topbeat:
+The `New()` function receives the configuration options defined for the Beat and
+creates a Beat object based on them. In our case the `Lsbeat` object has the following signature:
+
+[source,go]
+----------
+type Lsbeat struct {
+	done   chan struct{}
+	config config.Config
+	client publisher.Client
+}
+----------
+
+- `done`: Channel used by the `Run()` function to stop when the `Stop()` function is called.
+- `config`: Configuration options for the Beat
+- `client`: Publisher that takes care of sending the events to the
+  defined output.
+
+The configuration file of Lsbeat is `lsbeat.yml` and it's generated in the
+previous step:
 
 [source,yaml]
-----------------------------------------------------------------------
-input:
-  period: 1
+------------
+lsbeat:
+  # Defines how often an event is sent to the output
+  period: 10s
+------------
 
-  procs: [".*"]
+ - `period`: Defines how often to send out events
+
+In the case of `Lsbeat` there are two configuration options needed:
+ - `path`: The path of the directory to monitor
+ - `period`: How often should read the content of the directory
+
+
+[source,yaml]
+----------------------------------------------------------------------
+lsbeat:
+  # Defines how often an event is sent to the output
+  period: 10s
+  path: "."
 ----------------------------------------------------------------------
 
+The recommended way of handling the configuration is to create a
+`Config` structure with the configuration options and a `DefaultConfig` with
+the default configuration options. 
+
 And here are the corresponding Go structures, which are defined in `config.go`:
 
 [source,go]
 ----------------------------------------------------------------------
-type TopConfig struct {
-	Period *int64
-	Procs  *[]string
+package config
+
+import "time"
+
+type Config struct {
+	Period time.Duration `config:"period"`
+	Path   string        `config:"path"`
 }
 
-type ConfigSettings struct {
-	Input TopConfig
+var DefaultConfig = Config{
+	Period: 1 * time.Second,
+	Path:   ".",
 }
 ----------------------------------------------------------------------
 
 Pointers are used to distinguish between when the setting is completely
 missing from the configuration file and when it has a value that matches the
 type's default value.
 
-With these structures defined, the `Config` method looks like this:
-
+`Lsbeat` has the following `New()` function: 
 
 [source,go]
-----------------------------------------------------------------------
-func (tb *Topbeat) Config(b *beat.Beat) error {
-
-	err := cfgfile.Read(&tb.TbConfig, "") <1>
-	if err != nil {
-		logp.Err("Error reading configuration file: %v", err)
-		return err
+----------
+func New(b *beat.Beat, cfg *common.Config) (beat.Beater, error) {
+	config := config.DefaultConfig
+	if err := cfg.Unpack(&config); err != nil {
+		return nil, fmt.Errorf("Error reading config file: %v", err)
```

Comment:
fmt.Errorf not really required. The `err` contains config option that failed + source config option has been read from.